src/commands/tui/widgets/text_input.rs

```diff
@@ -40,22 +50,29 @@ impl ProcessEvent for TextInput {
     type Result = TextInputResult;
     fn input(&mut self, event: Event) -> TextInputResult {
         if let Event::Key(key) = event {
+            let KeyEvent {
+                code, modifiers, ..
+            } = key;
             use KeyCode::*;
-            match key {
-                KeyEvent { code: Esc, .. } => return TextInputResult::Cancel,
-                KeyEvent { code: Enter, .. } if self.lines == 1 => {
+            match (code, modifiers) {
+                (Esc, _) => return TextInputResult::Cancel,
+                (Char('q') | Char('x'), _) if !self.changeable => return TextInputResult::Cancel,
+                (Enter, _) if self.lines == 1 => {
                     return TextInputResult::Input(self.textarea.lines().join("\n"));
                 }
-                KeyEvent {
-                    code: Char('s'),
-                    modifiers: KeyModifiers::CONTROL,
-                    ..
-                } => {
+                (Char('s'), KeyModifiers::CONTROL) => {
                     return TextInputResult::Input(self.textarea.lines().join("\n"));
```

Comment:
for an input that isn't `changeable`, `Ctrl + s` will close the the `TextInput` popup. Which could be a bit misleading?
I'd advise that nothing should be done for an immutable popup when `Ctrl + s`

What do you think ?

Reply:
is changed, see below.